README.md

```diff
@@ -1,63 +1,70 @@
+## Table of Contents
 
-## Description
+- [What it's about (technical explanation)](#what-its-about-technical-explanation)
+- [Why you should use it?](#why-you-should-use-it)
+- [Usage (in a WordPress theme)](#usage-in-a-wordpress-theme)
+  - [Installation](#installation)
+  - [Setup](#setup)
+  - [Config Providers](#config-providers)
 
-The ContainerBuilder creates a [PSR-11](http://www.php-fig.org/psr/psr-11/) container, configured to use 
-[laminas-servicemanager](https://docs.laminas.dev/laminas-servicemanager) and 
-[laminas-di](https://docs.laminas.dev/laminas-di).
+## What it's about (technical explanation)
 
-## Usage
+This package allows you to quickly set up a dependency injection (DI) container and service providers within your project. No more, no less.
 
-```php
-<?php
+It creates a [PSR-11](http://www.php-fig.org/psr/psr-11/) container, configured to use [laminas-servicemanager](https://docs.laminas.dev/laminas-servicemanager) and [laminas-di](https://docs.laminas.dev/laminas-di).
 
-declare(strict_types=1);
+The typical use case this package was intended for is when building a WordPress theme or plugin where you want to glue together "packages" using [Composer](https://getcomposer.org/).
 
-if (!file_exists(__DIR__ . '/vendor/autoload.php')) {
-    throw new RuntimeException('Autoloader is not installed. Please run "composer install".');
-}
-require __DIR__ . '/vendor/autoload.php';
+It's not limited to WordPress though. That's just what we use it for. You can technically utilize it wherever you want.
 
-$container = (new \Woda\ContainerBuilder\ContainerBuilder([/* ... */]))
-    // Load files in folder with pattern {{,*.}global,{,*.}local}.php
-    ->withConfigFolder(__DIR__ . '/resources/config')
-    // Cache merged configuration in resources/cached-config.php
-    ->withCachedConfigFile(__DIR__ . '/resources/cached-config.php')
-    ->build();
+It's a very bare-bone implementation to solve one goal only: Setting up a DI container quickly and easily.
 
-// ... work with $container
-```
+## Why you should use it?
+
+While it may be daunting or seem like overhead at a first glance to use a DI container, it has many advantages. Once you clear the first hurdle of learning the basics, you will never want to go back doing things the *old* way.
+
+To counter your first concern: **No, this is not a framework.** This package is simply a ready-to-use implementation to get you going with writing clean and awesome quality-code quickly. What you make of it is up to you.
+
+But let's get back to the advantages of using a DI container:
+
+1. Firstly, you'll become a better developer. 🧐
+2. You will create cleaner and more readable code. 📖
+3. You will quickly learn to become much more productive by creating reusable packages instead of writing the same feature over and over again. 🚀
+4. You're working on a large project? It's going to be much easier to maintain it. 🗺️
+5. Your code will be more flexible because it is only loosely coupled. 💪
```

Comment:
Using DI won't change the coupling of your code, will it? 🤔

Reply:
Thank you for your feedback! We removed this line.